src/batch.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react'
+import classNames from 'classnames'
+
+class Batch extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      searching: false,
+      error: null,
+      results: null,
+      query: null
+    }
+  }
+
+  performSearch(query) {
+    if(!query) {
+      return
+    }
+    if (this.state.query === query) {
+      return
+    }
+
+    this.setState({
+      searching: true,
+      query: query
+    })
+
+    this.search(
+      query,
+      this.onResults.bind(this),
+      this.onError.bind(this))
+  }
+
+  search(query, success, error) {
+    $.ajax({
+      url: '/batch',
+      method: 'POST',
+      contentType: 'application/json',
+      data: JSON.stringify({
+        query_strings: query.terms,
+        collections: query.collections,
+      }),
+      success: success,
```

Comment:
Where are `success` and `error` defined? They're not in this file, so are they global?